src/imap/mod.rs

```diff
@@ -1730,16 +1730,15 @@ pub(crate) async fn set_uid_next(context: &Context, folder: &str, uid_next: u32)
 /// This method returns the uid_next from the last time we fetched messages.
 /// We can compare this to the current uid_next to find out whether there are new messages
 /// and fetch from this value on to get all new messages.
-async fn get_uid_next(context: &Context, folder: &str) -> u32 {
-    context
+async fn get_uid_next(context: &Context, folder: &str) -> Result<u32> {
```

Comment:
resultified this to prepare a bit better to a closed db (don't just return 0 on error)